runtime/src/main/java/org/corfudb/runtime/view/Address.java

```diff
@@ -2,14 +2,35 @@
 
 import lombok.Getter;
 
-/** Currently, this class holds static constants used
- * to represent addresses. In the future, it may replace the
- * use of long depending on performance.
+/**
+ * Address comprises of the global address in the log and the epoch associated with it.
+ * This class also holds static constants used to represent addresses.
  *
  * <p>Created by mwei on 1/6/17.</p>
  */
 public class Address {
 
+    @Getter
+    private final Long address;
+
+    @Getter
+    private final long epoch;
+
+    public Address(Long address, long epoch) {
+        this.address = address;
+        this.epoch = epoch;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        return epoch == ((Address)o).getEpoch();
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
This is to satisfy the cache on the logunit server. The cache would otherwise store multiple copies of the same address with different epochs. Even with the same epoch since the Address object is different. This would also cause a high cache miss rate. To overcome this, I changed the hashCode computation to be based on the address only.